src/sentry/utils/env.py

```diff
@@ -14,12 +15,22 @@ def gcp_project_id() -> str:
     if in_test_environment():
         return "__test_gcp_project__"
 
-    adc_path = os.environ.get("GOOGLE_APPLICATION_CREDENTIALS", "")
-    if adc_path:
-        with open(adc_path) as fp:
-            adc = json.load(fp)
-            if adc.get("quota_project_id") is not None:
-                return adc.get("quota_project_id")
+    # Try the metadata endpoint, if possible. If not, we'll assume a local environment as use the
+    # app credentials env variable instead.
+    try:
+        requests.get(
```

Comment:
where are you using the request response? should it return request.response

Reply:
but this seems like it will work both in single tenant and saas environments. just tried it

Reply:
Force pushed a fix here.